BuildingBlocks/src/DevelopmentKit.Identity/ValueObjects/IdentityAddress.cs

```diff
@@ -45,37 +47,38 @@ public static bool IsValid(string? stringValue)
 
         var lengthIsValid = stringValue.Length <= MAX_LENGTH;
 
-        var realm = stringValue[..3];
+        const string pattern = @"^did\:e\:((?:[a-zA-Z]+\.)+[a-zA-Z]+)\:dids\:(.+)(.{2})$";
+
+        var matches = Regex.Matches(stringValue, pattern, RegexOptions.IgnoreCase).First().Groups;
 
-        var concatenation = Base58.Bitcoin.Decode(stringValue.AsSpan(3)).ToArray();
-        var hashedPublicKey = concatenation[..20];
-        var givenChecksum = concatenation[20..];
+        if (matches is null) return false;
 
-        var realmBytes = Encoding.UTF8.GetBytes(realm);
-        var correctChecksum = CalculateChecksum(realmBytes, hashedPublicKey);
+        var givenChecksum = matches[3].Value;
+        var calculatedChecksum = CalculateChecksum(stringValue[..^2]);
```

Comment:
If you had a capturing group for the whole backbone specific part, you could use this group instead of having to get a substring.